gemini/sample-apps/fashion-trends-prediction/app/data_prep/dataProcessing.py

```diff
@@ -0,0 +1,378 @@
+                If there is no person in the image, return an empty list and only mention items that are visible in the image.
+                Use fashion terminology if possible and be verbose."""
+        """Input image:""",
+        fewshot_images[0],
+        """Output:
+    [
+        {
+            \"hat\": \"curved brim cotton red yankees baseball cap\",
+            \"sunglasses\": \"black small aviators \",
+            \"jacket\": \"black bomber waist length leather jacket\",
+            \"shirt\": \"white crew neck cotton t-shirt\",
+            \"pants\": \"cotton chinos full length straight leg beige pants\"
+        },
+        {
+            \"jacket\": \"cropped bomber green and white waist height  louis vuitton jacket\",
+            \"shirt\": \"white cotton crop top\",
+            \"skirt\": \"gabardine a-line beige mini skirt\",
+            \"socks\": \"white sheer ankle crew socks\",
+            \"bag\": \"green small rectangular leather top handle shoulder bag\"
+        }
+    ]
+
+
+    Input image:""",
+        fewshot_images[1],
+        """Output:
+[
+        {
+            \"top\": \"brown leopard print mid length cotton top with a square neckline and balloon sleeves and button-up front\",
+            \"jeans\": \"high-waisted,  mid-rise, straight-leg style, light-wash ripped distressed denim jeans\",
+            \"bag\": \"white leather rectangular tote bag\",
+            \"sunglasses\": \"black cat eye style square sunglasses\",
+            \"accessories\": \"choker gold necklace\"
+            \"accessories\": \"choker bangle  bracelet\"
+        }
+]
+
+    Input image:""",
+        user_image,
+        """Output:""",
+    ]
+
+    response = gemini_model.generate_content(
+        prompt,
+        generation_config={
+            "max_output_tokens": 2048,
+            "temperature": 0.4,
+            "top_p": 1,
+            "top_k": 32,
+        },
+        safety_settings={
+            generative_models.HarmCategory.HARM_CATEGORY_HATE_SPEECH: generative_models.HarmBlockThreshold.BLOCK_ONLY_HIGH,
+            generative_models.HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: generative_models.HarmBlockThreshold.BLOCK_ONLY_HIGH,
+            generative_models.HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT: generative_models.HarmBlockThreshold.BLOCK_ONLY_HIGH,
+            generative_models.HarmCategory.HARM_CATEGORY_HARASSMENT: generative_models.HarmBlockThreshold.BLOCK_ONLY_HIGH,
+        },
+        stream=False,
+    ).text
+
+    try:
+        response = json.loads(response)
+    except Exception as e:
+        print(e)
+        return {}
+
+    for json_ob in response:
+        for category in json_ob.keys():
+            if category in answer:
+                answer[category].append(json_ob[category])
+            else:
+                answer[category] = [json_ob[category]]
+
+    return answer
+
+
+def generateCaptionImageQnA(image_path):
```

Comment:
Add more comments and descriptive variable names to this function to make it more clear what is happening.

Reply:
This function is removed as it was using the imageQNA model which is now replaced with Gemini